model-optimizer/extensions/middle/PreserveRuntimeInfo.py

```diff
@@ -44,11 +45,37 @@ def preserve_rt_info(graph: Graph):
         for op in graph.get_op_nodes():
             op_name = op.soft_get('name', op.id)
             op_type = op.soft_get('type')
-            if op_type == 'Parameter' and op.has_valid('permute_attrs') and not op.has_and_set('nchw_layout'):
-                if not op.out_node(0).has_valid('permutation'):
-                    continue
-                permutation = op.out_node(0).permutation
-                if np.array_equal(permutation.inv, range(len(permutation.inv))):
+            order = None
+            if op_type == 'Parameter':
+                if 'auto_disable_nhwc_to_nchw' in graph.graph['cmd_params'] and \
+                        graph.graph['cmd_params'].auto_disable_nhwc_to_nchw:
+                    rank = op.out_port(0).data.get_shape().size
+                    if rank not in [4, 5]:
+                        continue
+                    order = list(range(rank))
+                    order.remove(1)
+                    order.append(1)
+                    order = int64_array(order)
+                elif op.has_valid('permute_attrs') and not op.has_and_set('nchw_layout') and \
+                        op.out_node(0).has_valid('permutation'):
+                    permutation = op.out_node(0).permutation
+                    order = permutation.inv
+                    if np.array_equal(order, range(len(permutation.inv))):
+                        continue
+
+                    # keep input in the framework format
+                    transpose = create_op_node_with_second_input(
+                        graph, Transpose, permutation.perm,
+                        {'name': op_name + '/Transpose({})'.format(permutation.perm)})
+
+                    # source mode is used to keep tensor names at Parameter node
+                    op.out_port(0).get_connection().insert_node(transpose, "source")
+
+                    if op.has_valid('permute_attrs'):
+                        del op['permute_attrs']
+                    if op.out_node(0).has_valid('permutation'):
+                        del op.out_node(0)['permutation']
```

Comment:
@popovaan, could you please double-check your part here?

Reply:
Checked. Looks correct.